validator/coordinator/src/test/java/tech/pegasys/artemis/validator/coordinator/ValidatorCoordinatorTest.java

```diff
@@ -13,54 +13,115 @@
 
 package tech.pegasys.artemis.validator.coordinator;
 
+import static org.assertj.core.api.Assertions.assertThat;
 import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.atLeastOnce;
 import static org.mockito.Mockito.doReturn;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.never;
 import static org.mockito.Mockito.spy;
 import static org.mockito.Mockito.verify;
+import static tech.pegasys.artemis.datastructures.blocks.BeaconBlockBodyLists.createAttestations;
 
 import com.google.common.eventbus.EventBus;
+import com.google.common.primitives.UnsignedLong;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import tech.pegasys.artemis.datastructures.blocks.BeaconBlock;
+import tech.pegasys.artemis.datastructures.util.MockStartValidatorKeyPairFactory;
 import tech.pegasys.artemis.statetransition.AttestationAggregator;
 import tech.pegasys.artemis.statetransition.BeaconChainUtil;
 import tech.pegasys.artemis.statetransition.BlockAttestationsPool;
 import tech.pegasys.artemis.statetransition.events.BroadcastAttestationEvent;
 import tech.pegasys.artemis.storage.ChainStorageClient;
+import tech.pegasys.artemis.storage.events.SlotEvent;
+import tech.pegasys.artemis.util.Waiter;
+import tech.pegasys.artemis.util.bls.BLSKeyPair;
 import tech.pegasys.artemis.util.config.ArtemisConfiguration;
 
 public class ValidatorCoordinatorTest {
 
+  private BlockAttestationsPool blockAttestationsPool;
+  private AttestationAggregator attestationAggregator;
+  private EventBus eventBus;
+  private ChainStorageClient storageClient;
+  private ArtemisConfiguration config;
   private ValidatorCoordinator vc;
 
-  @Test
-  void noValidators_noAttestationProduced() throws Exception {
-    int numValidators = 0;
+  private final int numValidators = 12;
 
-    ArtemisConfiguration config = mock(ArtemisConfiguration.class);
+  @BeforeEach
+  void setup() {
+    config = mock(ArtemisConfiguration.class);
     doReturn(0).when(config).getNaughtinessPercentage();
     doReturn(numValidators).when(config).getNumValidators();
     doReturn(null).when(config).getValidatorsKeyFile();
     doReturn(0).when(config).getInteropOwnedValidatorStartIndex();
     doReturn(numValidators).when(config).getInteropOwnedValidatorCount();
```

Comment:
These can just use the `when(config.getWhatever()).thenReturn(value);` approach.  `doReturn` is only needed when you're using a spy and the method has side effects or a previous `when` setup a mock to throw an exception.

Reply:
I like using `doReturn` by default because if you end up needing to refactor (and the mocks become spies or whatever), it still works.  Any reason not to always use this one style?

Reply:
done.

Reply:
Type-safety is the big reason to avoid `doReturn` unless you really have to use it. From the mockito javadoc for `doReturn`:
```
* <b>Beware that {@link Mockito#when(Object)} is always recommended for stubbing because it is argument type-safe
     * and more readable</b> (especially when stubbing consecutive calls).
```